tests/pipelines/pipeline/shared/transforms/test_validate_csv.py

```diff
@@ -0,0 +1,106 @@
+    fixture_file = Path(fixtures_files_dir / 'test_validate_csv_data.csv')
+    columns = ['OBS_VALUE', 'OBS_STATUS']
+    expected_len = 5 # fixture file is a small file with only 5 columns
+    
+    with pytest.raises(AssertionError) as err:
+        _correct_columns_exist(fixture_file, columns)
+        
+    assert f"Number of columns in csv file ({expected_len}) does not match len of expected columns ({len(columns)})" in str(err.value)
+    
+def test_correct_columns_exist_different_column_name():
+    fixture_file = Path(fixtures_files_dir / 'test_validate_csv_data.csv')
+    incorrect_column = 'Incorrect_column_name'
+    columns = ['ID', 'TIME_PERIOD', 'OBS_VALUE', 'OBS_STATUS', incorrect_column]
+    
+    with pytest.raises(AssertionError) as err:
+        _correct_columns_exist(fixture_file, columns)
+        
+    assert f"Expect column {incorrect_column} not found in csv file" in str(err.value)
+
+def test_dataframe_has_no_blanks_optional_columns_not_a_list():
+    optional_columns = 'col1'
+    df = pd.DataFrame({'col1': ['1', '2'], 'col2': ['3', '4']})
+
+    with pytest.raises(AssertionError) as err:
+        _dataframe_has_no_blanks(df, check_specific_columns=optional_columns)
+
+    assert "check_specific_columns kwarg must be a list" in str(err.value)
+
+def test_dataframe_has_no_blanks_optional_columns_incorrect():
+    optional_columns = ['col3']
+    df = pd.DataFrame({'col1': ['1', '2'], 'col2': ['3', '4']})
+
+    with pytest.raises(AssertionError) as err:
+        _dataframe_has_no_blanks(df, check_specific_columns=optional_columns)
+
+    assert f"{optional_columns[0]} in check_specific_columns not found in dataframe" in str(err.value)
+    
+def test_dataframe_has_no_blanks_whitespace_entry():
+    blank_value = '     '
+    df = pd.DataFrame({'col1': ['1', blank_value], 'col2': ['3', '4']})
+    column = 'col1'
+    
+    with pytest.raises(ValueError) as err:
+        _dataframe_has_no_blanks(df)
+        
+    assert str(err.value) == f"{column} has blank entries"
+
+def test_dataframe_has_no_blanks_none_entry():
+    blank_value = None
+    df = pd.DataFrame({'col1': ['1', blank_value], 'col2': ['3', '4']})
+    column = 'col1'
+    
+    with pytest.raises(ValueError) as err:
+        _dataframe_has_no_blanks(df)
+        
+    assert str(err.value) == f"{column} has blank entries"
+
+def test_dataframe_has_no_duplicates():
+    df = pd.DataFrame({'col1': ['1', '2', '1'], 'col2': ['3', '4', '3']}) # rows 1 & 3 the same
+    
+    with pytest.raises(AssertionError) as err:
+        _dataframe_has_no_duplicates(df)
+        
+    assert "Found duplicate rows in the dataframe, failed validation" in str(err.value)
+
+def test_generated_dataframe_slices():
+    fixture_file = Path(fixtures_files_dir / 'test_validate_csv_data.csv')
+
+    try:
+        for df in generated_dataframe_slices(fixture_file, chunk_size=2):
+            assert type(df) == pd.DataFrame, "generated_dataframe_slices not returning a pd.DataFrame"
```

Comment:
you should really test that its returning a dataset with a number of rows equal to what you've asked for as well